app/scripts/controllers/alert-controller.test.ts

```diff
@@ -28,231 +29,288 @@ const EMPTY_ACCOUNT = {
+  | [WithControllerOptions, WithControllerCallback<ReturnValue>];
 
-  beforeEach(() => {
-    controllerMessenger = new ControllerMessenger<
-      AllowedActions,
-      AllowedEvents
-    >();
-    controllerMessenger.registerActionHandler(
-      'AccountsController:getSelectedAccount',
-      () => EMPTY_ACCOUNT,
-    );
+async function withController<ReturnValue>(
+  ...args: WithControllerArgs<ReturnValue>
+): Promise<ReturnValue> {
+  const [{ ...rest }, fn] = args.length === 2 ? args : [{}, args[0]];
+  const { ...alertControllerOptions } = rest;
 
-    const alertMessenger = controllerMessenger.getRestricted({
+  const controllerMessenger = new ControllerMessenger<
+    AllowedActions | AlertControllerGetStateAction,
+    AllowedEvents | AlertControllerStateChangeEvent
+  >();
+
+  const alertControllerMessenger: AlertControllerMessenger =
+    controllerMessenger.getRestricted({
       name: 'AlertController',
-      allowedActions: [`AccountsController:getSelectedAccount`],
-      allowedEvents: [`AccountsController:selectedAccountChange`],
+      allowedActions: ['AccountsController:getSelectedAccount'],
+      allowedEvents: ['AccountsController:selectedAccountChange'],
     });
 
-    alertController = new AlertController({
-      state: {
-        unconnectedAccountAlertShownOrigins: {
-          testUnconnectedOrigin: false,
-        },
-        web3ShimUsageOrigins: {
-          testWeb3ShimUsageOrigin: 0,
-        },
-      },
-      controllerMessenger: alertMessenger,
-    });
+  controllerMessenger.registerActionHandler(
+    'AccountsController:getSelectedAccount',
+    jest.fn().mockReturnValue(EMPTY_ACCOUNT),
+  );
+
+  const controller = new AlertController({
+    messenger: alertControllerMessenger,
+    state: getDefaultAlertControllerState(),
+    ...alertControllerOptions,
+  });
+
+  return await fn({
+    controller,
+    messenger: controllerMessenger,
   });
+}
 
+describe('AlertController', () => {
   describe('default state', () => {
-    it('should be same as AlertControllerState initialized', () => {
-      expect(alertController.store.getState()).toStrictEqual({
-        alertEnabledness: {
-          unconnectedAccount: true,
-          web3ShimUsage: true,
-        },
-        unconnectedAccountAlertShownOrigins: {
-          testUnconnectedOrigin: false,
-        },
-        web3ShimUsageOrigins: {
-          testWeb3ShimUsageOrigin: 0,
-        },
+    it('should be same as AlertControllerState initialized', async () => {
+      await withController({ state: initialState }, ({ controller }) => {
+        expect(controller.state).toStrictEqual({
+          alertEnabledness: {
+            unconnectedAccount: true,
+            web3ShimUsage: true,
+          },
+          unconnectedAccountAlertShownOrigins: {
+            testUnconnectedOrigin: false,
+          },
+          web3ShimUsageOrigins: {
+            testWeb3ShimUsageOrigin: 0,
+          },
+        });
```

Comment:
Hmm, this test is funny. "Default state" implies that no initial state is passed and yet this controller was instantiated with initial state in the test setup. And now we are explicitly passing state. So I feel like one or both of these things isn't needed.

Perhaps we can simplify it to the following?

``` suggestion
      const defaultState = getDefaultAlertControllerState();

      await withController({ state: defaultState }, ({ controller }) => {
        expect(controller.state).toStrictEqual(defaultState);